### ISSUE TITLE
1f Noise generation deadlocked

### ISSUE THREAD

**Issue by R0ckySu** [stephanlphilips/DM_solver#7]
1f noise generation takes unusually long time for finishing FFT(for a 500000*1 vector, Matlab takes less than 1s, but DM_solver takes minutes long without finishing).

Guessing it might be originated from irregular size of the vector we put in to FFT

Temporary solution:

makes the length of vector be the multiple of 10
```cpp
	int add_on_steps = floor(std::abs(tmp(generator))/10) * 10;
	new_steps += add_on_steps;
```

**Comment by stephanlphilips** [comment]
Yes, the size should indeed be regular, to be specific, 2**n. At the moment we are working at a new version where this will be fixed.
